frontend/src/providers/GemProvider.vue

```diff
@@ -57,7 +81,7 @@ export default defineComponent({
         if (err instanceof Error) {
           return Promise.reject(err);
         }
-        return Promise.reject(new Error("Unknown error type"));
+        return Promise.reject(new Error((err as any).message));
```

Comment:
Why not just `Promise.reject(err)`?

Reply:
right lol. i think i had some typescript issues which was why I had the above. But Promise.reject(err) works great!